Add an optional mail forwarder used to accept mail sent to an AWS hosted domain and forward it to an external recipient or recipients. (#128)
schubergphilis/terraform-aws-mcaf-landing-zone: README.md

# Detailed configuration

> ## AWS SES Accounts mail alias
> By setting the `ses_root_accounts_mail_alias` variable you can enable secure mailboxes/IT service catalog aliases for all root accounts and forward it to an external recipient or recipients. Emails are received via AWS Simple Email Service (SES) and forwarded to an email forwarder lambda which sends the email to the destination email server as specified in the `recipient_mapping` variable of `ses_root_accounts_mail_alias`.

Reviewer: ```suggestion
## AWS Simple Email Service (SES)

Setting the `ses_root_accounts_mail_alias` variable will configure SES to accept email for one or more email addresses using a pre-existing domain. This email address or addresses can be configured to forward mail it receives to external recipients. It also supports plus-style email aliases which allows you to configure one SES email address, such as `root@aws.yourcompany.com` which forwards mail to `mailbox@yourcompany.com`, and create new AWS root accounts using aliases, such as `root+dev@aws.yourcompany.com` or `root+prod@aws.yourcompany.com`.

Learn more how this works by looking at the [terraform-aws-mcaf-ses-forwarder module](https://github.com/schubergphilis/terraform-aws-mcaf-ses-forwarder).
```

Before setting the `ses_root_accounts_mail_alias` variable, make sure that an AWS Route53 hosted zone is created. For example aws.yourcompany.com. Pass this domain using the `domain` variable of `ses_root_accounts_mail_alias`.

Example:

```hcl

> ses_root_accounts_mail_alias = {
>   domain   = "aws.yourcompany.com"
>   from_email = "int@aws.yourcompany.com"
>   recipient_mapping = {
>     "int@aws.yourcompany.com" = [
>       "inbox@yourcompany.com"
>     ]
>   }
> }

Reviewer: ```suggestion
ses_root_accounts_mail_alias = {
  domain     = "aws.yourcompany.com"
  from_email = "root@aws.yourcompany.com"

  recipient_mapping = {
    "root@aws.yourcompany.com" = ["mailbox@yourcompany.com"]
  }
}
```

```

> By default, you have to create the email addresses for the accounts created using the [MCAF Account Vending Machine (AVM) module](https://github.com/schubergphilis/terraform-aws-mcaf-avm) yourself. Using this functionality you can pass aliases of the mailbox created. E.g. int+\<account-name\>@aws.yourcompany.com. 

Reviewer: ```suggestion

By default, you have to create the email addresses for the accounts created using the [MCAF Account Vending Machine (AVM) module](https://github.com/schubergphilis/terraform-aws-mcaf-avm) yourself. Using this functionality you can pass aliases of the mailbox created. E.g. int+\<account-name\>@aws.yourcompany.com. 
```

# Detailed configuration

> ## AWS SES Root Accounts mail forwarder

Reviewer: ```suggestion
## AWS SES Root Accounts mail forwarder

```

Setting the `ses_root_accounts_mail_forward` variable creates the necessary AWS Simple Email Service (SES) resources to accept mail sent to an AWS hosted domain and forward it to an external recipient or recipients. This can be used to enable secure mailboxes/IT service catalog aliases for all root accounts. Emails are received via AWS SES and forwarded to an email forwarder lambda which sends the email to the destination email server as specified in the `recipient_mapping` variable of `ses_root_accounts_mail_forward`.

Before setting the `ses_root_accounts_mail_forward` variable, make sure that an AWS Route53 hosted zone is created. For example aws.yourcompany.com. Pass this domain using the `domain` variable of `ses_root_accounts_mail_forward`.

Example:

```hcl
ses_root_accounts_mail_forward = {

>   domain   = "aws.yourcompany.com"

Reviewer: ```suggestion
  domain     = "aws.yourcompany.com"
```

  from_email = "root@aws.yourcompany.com"

  recipient_mapping = {
    "root@aws.yourcompany.com" = [
      "inbox@yourcompany.com"
    ]
  }

> }

Reviewer: ```suggestion
}

```

> ```

Reviewer: ```suggestion

```
```
